File: examples/gradle-ktor/src/main/kotlin/community/flock/wirespec/example/gradle/app/todo/TodoService.kt
package community.flock.wirespec.example.gradle.app.todo

interface TodoService : HasTodoRepository

fun TodoService.getAllTodos() = todoRepository.getAllTodos()

fun TodoService.getTodoById(id: Todo.Id) = todoRepository.getTodoById(id)

fun TodoService.saveTodo(todo: Todo) = todoRepository.saveTodo(todo)

fun TodoService.deleteTodoById(id: Todo.Id) = todoRepository.deleteTodoById(id)

File: examples/scala-zio/src/main/scala/example/port/TodoService.scala
package example.port

import community.flock.wirespec.generated.model.{PotentialTodoDto, TodoDto, TodoId}
import zio.*

trait TodoService:
  def getTodos(done: Option[Boolean]): Task[List[TodoDto]]
  def createTodo(potential: PotentialTodoDto): Task[TodoDto]
  def getTodoById(id: TodoId): Task[Option[TodoDto]]
  def deleteTodoById(id: TodoId): Task[Option[TodoDto]]
